src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs

```diff
@@ -77,9 +83,8 @@ IInternalServiceHubContext IInternalServiceHubContext.WithEndpoints(IEnumerable<
                 //add required service instances
                 .AddSingleton(ServiceProvider.GetRequiredService<IOptions<ServiceManagerOptions>>())
                 .AddSingleton(_negotiateProcessor)
-                .AddSingleton(_endpointManager)
-
-                .BuildServiceProvider()
+                .AddSingleton(_endpointManager);
+                return services.BuildServiceProvider()
```

Comment:
Add an empty line before `return` and remove extra spaces before `return`. Also remove extra empty lines in the `ServiceConnection()` call.